presto-jdbc/src/main/java/com/facebook/presto/jdbc/ColumnInfo.java

```diff
@@ -133,11 +134,23 @@ public static void setTypeInfo(Builder builder, TypeSignature type)
                 builder.setScale(0);
                 builder.setColumnDisplaySize(24);
                 break;
+            case "char":
+                builder.setSigned(false);
+                builder.setPrecision(type.getParameters().get(0).getLongLiteral().intValue());
+                builder.setColumnDisplaySize(type.getParameters().get(0).getLongLiteral().intValue());
+                builder.setScale(0);
+                break;
             case "varchar":
-                builder.setSigned(true);
-                builder.setPrecision(VARCHAR_MAX);
+                builder.setSigned(false);
+                if (type.getParameters().get(0).getLongLiteral().intValue() > VarcharType.MAX_LENGTH) {
```

Comment:
We can simplify this a bit using Math.max()

Reply:
Done.